htdocs/accountancy/journal/bankjournal.php

```diff
@@ -90,17 +90,52 @@
 
 $error = 0;
 
-$year_current = strftime("%Y", dol_now());
-$pastmonth = strftime("%m", dol_now()) - 1;
-$pastmonthyear = $year_current;
-if ($pastmonth == 0) {
-	$pastmonth = 12;
-	$pastmonthyear--;
-}
-
 $date_start = dol_mktime(0, 0, 0, $date_startmonth, $date_startday, $date_startyear);
 $date_end = dol_mktime(23, 59, 59, $date_endmonth, $date_endday, $date_endyear);
 
+// Period by default on transfer (0: previous month | 1: current month | 2: fiscal year)
```

Comment:
All code block is very similar on all pages.
I suggest to move it into function  
getDefaultDatesForTransfer() into accounting.lib.php
that return array('date_start'=>,  'date_end'=> ,  'month_start'=> ...